django/db/migrations/graph.py

```diff
@@ -207,8 +221,18 @@ def remove_replacement_node(self, replacement, replaced):
         nodes are not remapped as they are expected to be correct already.
         """
         self.nodes.pop(replacement, None)
-        # Pop without `None` here as the replacement node must exist.
-        replacement_node = self.node_map.pop(replacement)
+        try:
+            replacement_node = self.node_map.pop(replacement)
+        except KeyError as exc:
+            exc_value = NodeNotFoundError(
+                "Unable to remove replacement node %r. It was either never added"
+                " to the migration graph, or has been removed already." % (replacement, ),
+                replacement
+            )
+            exc_value.__cause__ = exc
+            if not hasattr(exc, '__traceback__'):
+                exc.__traceback__ = sys.exc_info()[2]
+            six.reraise(NodeNotFoundError, exc_value, sys.exc_info()[2])
```

Comment:
@MarkusH Having to write all this just to raise an error for a situation that should technically never happen feels awfully verbose. Do you know of a better way?